### PR TITLE
Refactor control state mixin: use new mixins

### LINKED ISSUE #20: Refactor control state mixin to use new mixins
Change `ControlStateMixin` to use the new mixins:

- `DisabledStateMixin` - #10 
- `FocusVisibleMixin` - #11 

Note: this would need `0.2.0` version bump.

### PR DESCRIPTION
**Pull request by web-padawan**
Fixes #20

### CODE REVIEW

File: packages/control-state-mixin/control-state-mixin.ts [hunk 1]
```diff
@@ -110,26 +65,36 @@ export const ControlStateMixin = <T extends LitBase>(base: T): LitControlState &
...
+          if (tabIndex !== -1) {
+            this._previousTabIndex = tabIndex;
+          }
+          // We need this because of resetting tabIndex to null below,
+          // so in `updated()` we have a reference to the old value.
+          props.set('tabIndex', tabIndex);
+
+          // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+          // @ts-ignore
+          this.tabIndex = null;
```
Comment (Haprog) [inline packages/control-state-mixin/control-state-mixin.ts]: Feels counter-intuitive to need these 2 comment lines. Should we just globally disable `@typescript-eslint/ban-ts-ignore` rule if we sometimes need `@ts-ignore`?

File: packages/control-state-mixin/control-state-mixin.ts [hunk 2]
```diff
@@ -145,33 +110,29 @@ export const ControlStateMixin = <T extends LitBase>(base: T): LitControlState &
...
+      } else if (path.indexOf(this.focusElement) !== -1 && !this.disabled && super._onFocusin) {
+        super._onFocusin(event);
+      }
     }
 
-    private _bodyKeyupListener() {
-      this._tabPressed = false;
+    protected _setFocused(focused: boolean) {
+      super._setFocused && super._setFocused(focused); // eslint-disable-line no-unused-expressions
     }
```
Comment (Haprog) [inline packages/control-state-mixin/control-state-mixin.ts]: Is this really needed when it only calls super?

Comment (web-padawan) [reply packages/control-state-mixin/control-state-mixin.ts]: The purpose of this method was to only check for `super` in one place.
I agree it looks a bit weird but IMO it's easier to keep it this way.

File: packages/control-state-mixin/control-state-mixin.ts (hunk 1 above)
Comment (web-padawan) [reply packages/control-state-mixin/control-state-mixin.ts]: Thanks, that makes sense. Done.

File: packages/control-state-mixin/control-state-mixin.ts (hunk 2 above)
Comment (web-padawan) [reply packages/control-state-mixin/control-state-mixin.ts]: Discussed in person and agreed to change so we don't have `eslint-disable` comments.

**Pull request merged**